java/src/com/google/idea/blaze/java/run/hotswap/BlazeReloadFileAction.java

```diff
@@ -170,17 +191,25 @@ private void findAndCopyOutputFile(VirtualFile sourceFile, File tempClassDir, St
                         ZipEntry entry;
                         while ((entry = jis.getNextEntry()) != null) {
                             if (entry.isDirectory() && entry.getName().equals(jarDirectory)) {
-                                while ((entry = jis.getNextEntry()) != null && !entry.isDirectory()) {
-                                    String entryFileName = entry.getName().substring(entry.getName().lastIndexOf(Paths.DELIM) + 1);
-                                    String entryFileNameWithoutExtention = entryFileName.contains(".") ? entryFileName.substring(0, entryFileName.lastIndexOf('.')) : entryFileName;
-                                    if (entryFileNameWithoutExtention.startsWith(filenameWithoutExtension) &&
-                                            (entryFileNameWithoutExtention.length() == filenameWithoutExtension.length() ||
-                                                    INNER_CLASS_PATTERN.matcher(entryFileNameWithoutExtention.substring(filenameWithoutExtension.length())).matches())) {
-                                        File out = new File(tempClassDir, entryFileName);
-                                        Files.copy(jis, out.toPath(), StandardCopyOption.REPLACE_EXISTING);
-
+                                LOGGER.info(String.format("Found entries for target package %s in %s", jarDirectory, entry.getName()));
+                                int scannedEntries = 0;
+                                int foundEntries = 0;
+                                while ((entry = jis.getNextEntry()) != null && entry.getName().startsWith(jarDirectory)) {
+                                    scannedEntries++;
+                                    if (!entry.isDirectory()) {
+                                        String entryFileName = entry.getName().substring(entry.getName().lastIndexOf(Paths.DELIM) + 1);
```

Comment:
In theory, we should know the index since it will be jarDirectory.length()

Reply:
This is your code, I haven't touched this ;)

Reply:
Lol fair